### PR TITLE
React delete category

### PR DESCRIPTION
**Pull request by cynthia-doc**
Delete category using drag and drop, post delete call to db

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,35 @@
...
+### Step 1: Start Server API Node on Port 8080
+Connecting to Our Remote MongoDB Atlas 
+- Create a file in ./server named '.env' for credential setup. (Reach out for credential details!)
+
+Connecting to your own MongoDB cluster or local MongoDB
+- Navigate to ./server/db/index.js
+- Change <code>defaultUri</code> to your own connection key
+
+Start Server
+- <code>$ cd ./server</code>
```
Comment (Anthony-Fiddes) [inline README.md:17]: Nit: I know this isn't the place for this, but doesn't code go in triple `?

File: server/test/suites/itemRouter.test.js [hunk 2]
```diff
@@ -18,6 +18,12 @@ const expect = chai.expect;
 chai.use(chaiHttp);
 
 // TODO: Mock documents used for testing
+let item1;
```
[the remark this answers is not in the document]
Comment (cynthia-doc) [reply server/test/suites/itemRouter.test.js]: My bad! it previously wasn't based off of master but now it is, and these files are not changes on this PR

File: webapp-client/src/component/DragContainer.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    return (
+      <>
+        <Drag
+          title={this.props.title}
+          id={this.props.id}
+          deleteCategory={this.deleteCategory}
+          setCurrentCategory={this.props.setCurrentCategory}
+          currentCategoryId={this.props.currentCategoryId}
+        />
+        {this.renderErrorMessage('Error deleting category')}
```
[the remark this answers is not in the document]
Comment (cynthia-doc) [reply webapp-client/src/component/DragContainer.js:93]: this is in errorify.js, which I imported as allowErrorMessage :)

File: webapp-client/src/component/CategoryList.js [hunk 4]
```diff
@@ -31,42 +34,47 @@ export default class CategoryList extends React.Component {
         categoryList,
       });
     } catch (error) {
-      this.showErrorMessage();
+      this.setState({ hasError: true });
```
Comment (Anthony-Fiddes) [inline webapp-client/src/component/CategoryList.js:37]: Why are you opting to set the state manually?

File: webapp-client/src/component/CategoryList.js [hunk 5]
```diff
@@ -31,42 +34,47 @@ export default class CategoryList extends React.Component {
...
+        />
+      </>
     );
   }
 
+  clearHasError() {
+    this.setState({
+      hasError: false,
+    });
+  }
```
Comment (Anthony-Fiddes) [inline webapp-client/src/component/CategoryList.js:67]: is errorify not working?

File: webapp-client/public/index.html [hunk 6]
```diff
@@ -1,25 +1,30 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <link
+      href="%PUBLIC_URL%/fontawesome-free-5.13.1-web/css/all.css"
+      rel="stylesheet"
+    />
+    <link rel="icon" href="%PUBLIC_URL%/lobster_icon.jpg" />
+    <link rel="stylesheet" href="%PUBLIC_URL%/bulma.min.css" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
```
Comment (Sherry2001) [inline webapp-client/public/index.html:12]: #000

File: webapp-client/src/component/DeleteCategory.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import ErrorMessage from './ErrorMessage';
+
+export default function DeleteCategory (props) {
+  let hasError = false;
+
+  const deleteCategory = async () => {
+    console.log("in deleteCategory");
+    const request = {
+      method: 'POST',
```
Comment (Sherry2001) [inline webapp-client/src/component/DeleteCategory.js:10]: this is a "DELETE" method

File: webapp-client/src/component/DeleteCategory.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    hasError= false;
+  };
+
+  return (
+    <>
+      <ErrorMessage
+        hasError={hasError}
+        message={'Error deleting category'}
+        closePopup={clearHasError}
+      />
```
Comment (Anthony-Fiddes) [inline webapp-client/src/component/DeleteCategory.js:39]: Same as above

File: webapp-client/src/component/DragContainer.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        categoryId: this.props.id,
+      }),
+      headers: { 'Content-type': 'application/json' },
+    };
+    try {
+      const response = await fetch(
+        process.env.REACT_APP_API_URL + '/category/deleteCategory',
+        request
+      );
+      if (response.status !== 200) {
```
Comment (Sherry2001) [inline webapp-client/src/component/DragContainer.js:75]: how come this is here but also in deletecategory.js?

File: webapp-client/src/component/CategoryList.js (hunk 4 above)
Comment (cynthia-doc) [reply webapp-client/src/component/CategoryList.js:37]: ah I don't think this has master pulled.

File: webapp-client/src/component/DragContainer.js (hunk 9 above)
Comment (cynthia-doc) [reply webapp-client/src/component/DragContainer.js:75]: deletecategory is a file i initially had but it's extra now, and it's deleted not:))

**Pull request merged**